src/Microsoft.SqlTools.ServiceLayer/AutoParameterizaition/SqlParameterizer.cs

```diff
@@ -0,0 +1,128 @@
+
+namespace Microsoft.SqlTools.ServiceLayer.AutoParameterizaition
+{
+    public static class SqlParameterizer
+    {
+        private const int maxStringLength = 300000; // Approximately 600 Kb
+        private static readonly IList<ScriptFileMarker> EmptyCodeSenseItemList = Enumerable.Empty<ScriptFileMarker>().ToList();
+
+        public static SqlScriptGenerator GetScriptGenerator()
+        {
+            return new Sql150ScriptGenerator();
+        }
+
+        public static TSqlParser GetTSqlParser(bool initialQuotedIdentifiers)
+        {
+            return new TSql150Parser(initialQuotedIdentifiers);
+        }
+
+        /// <summary>
+        /// This method will parameterize the given SqlCommand.
+        /// Any single literal on the RHS of a declare statement will be parameterized
+        /// Any other literals will be ignored
+        /// </summary>
+        /// <param name="commandToParameterize">Command that will need to be parameterized</param>
+        public static void Parameterize(this DbCommand commandToParameterize)
+        {         
+            TSqlFragment rootFragment = GetAbstractSyntaxTree(commandToParameterize);
+            TsqlMultiVisitor multiVisitor = new TsqlMultiVisitor(isCodeSenseRequest: false); // Use the vistor pattern to examine the parse tree
+            rootFragment.AcceptChildren(multiVisitor); // Now walk the tree
+
+            //reformat and validate the transformed command
+            SqlScriptGenerator scriptGenerator = GetScriptGenerator();
+            scriptGenerator.GenerateScript(rootFragment, out string formattedSQL);
+
+            commandToParameterize.CommandText = formattedSQL;
+            commandToParameterize.Parameters.AddRange(multiVisitor.Parameters.ToArray());
+
+            multiVisitor.Reset();
+        }
+
+        /// <summary>
+        /// Parses the given script to provide message, warning, error.
+        /// </summary>
+        /// <param name="scriptToParse">Script that will be parsed</param>
+        /// <param name="telemetryManager">Used to emit telemetry events</param>
+        /// <returns></returns>
+        public static IList<ScriptFileMarker> CodeSense(string scriptToParse)
+        {
+            if (scriptToParse == null)
+            {
+                return EmptyCodeSenseItemList;
+            }
+
+            int CurrentScriptlength = scriptToParse.Length;
+            if (CurrentScriptlength > maxStringLength)
+            {
+                ScriptFileMarker maxStringLengthCodeSenseItem = new ScriptFileMarker
+                {
+                    Level = ScriptFileMarkerLevel.Error,
+                    Message = SR.ScriptTooLarge(maxStringLength, CurrentScriptlength),
+                    ScriptRegion = new ScriptRegion
+                    {
+                        StartLineNumber = 1,
+                        StartColumnNumber = 1,
+                        EndLineNumber = 2,
+                        EndColumnNumber = 1
```

Comment:
Are these the default values for when a string is too long again?  Add a comment explaining what they are, or someone might think `EndLineNumber = 2` is a mistake given the rest are 1

Reply:
added comment to explain magic numbers